src/script/interpreter.cpp

```diff
@@ -1681,6 +1750,218 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_INSPECTINPUTOUTPOINT:
+                case OP_INSPECTINPUTASSET:
+                case OP_INSPECTINPUTVALUE:
+                case OP_INSPECTINPUTSCRIPTPUBKEY:
+                case OP_INSPECTINPUTSEQUENCE:
+                case OP_INSPECTINPUTISSUANCE:
+                {
+                    // Input inspection opcodes only available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    if (stack.size() < 1)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    int idx = CScriptNum(stacktop(-1), fRequireMinimal).getint();
+                    popstack(stack);
+
+                    auto inps = checker.GetTxvIn();
+                    const PrecomputedTransactionData *cache = checker.GetPrecomputedTransactionData();
+                    // Return error if the evaluation context is unavailable
+                    // TODO: Handle accoding to MissingDataBehavior
+                    if (!inps || !cache)
+                        return set_error(serror, SCRIPT_ERR_INTROSPECT_CONTEXT_UNAVAILABLE);
+                    auto spent_outputs = cache->m_spent_outputs;
+                    if (idx < 0 || static_cast<unsigned int>(idx) >= inps->size() || static_cast<unsigned int>(idx) >= spent_outputs.size())
```

Comment:
spent_outputs.size should be equal to inputs size.  Maybe you should be asserting that.

Reply:
Actually we also need an return false if cache.m_spent_outputs_ready is false, with a TODO to make it assert based on `mdb` value.

See for example, https://github.com/bitcoin/bitcoin/blob/eb09c26724e3f714b613788fc506f2ff3a208d2c/src/script/interpreter.cpp#L1523-L1525

Reply:
Adding an assert after checking `m_spent_outputs`

Reply:
Why not add an assert before the `if` statement and remove one of the clauses?